R/zchunk_L142.pfc_R_S_T_Y.R

```diff
@@ -8,83 +8,179 @@
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    EDGAR_sector <- get_data(all_data, "emissions/EDGAR/EDGAR_sector_fgas")
+    GWP <- get_data(all_data, "emissions/A41.GWP")
+
+    get_data(all_data, "temp-data-inject/L144.in_EJ_R_bld_serv_F_Yh")  %>%
+      gather(year, value, -GCAM_region_ID, -sector, -fuel, -service) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L144.in_EJ_R_bld_serv_F_Yh
+    get_data(all_data, "emissions/EDGAR/EDGAR_SF6")  %>%
+      gather(year, value, -IPCC_Annex, -World_Region, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(year = as.integer(substr(year, 1, 5))) ->
+      EDGAR_SF6
+    get_data(all_data, "emissions/EDGAR/EDGAR_C2F6") %>%
+      gather(year, value, -IPCC_Annex, -World_Region, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(year = as.integer(substr(year, 1, 5))) ->
+      EDGAR_C2F6
+    get_data(all_data, "emissions/EDGAR/EDGAR_CF4") %>%
+      gather(year, value, -IPCC_Annex, -World_Region, -ISO_A3, -Name, -IPCC, -IPCC_description) %>%
+      mutate(year = as.integer(substr(year, 1, 5))) ->
+      EDGAR_CF4
+
+    # This chunk maps EDGAR HFC emissions to GCAM technologies
+    # First, create a table with all EDGAR HFCs, and prep "Other_F" table for use by renaming column.
+
+    EDGAR_SF6 %>%
+      mutate(Non.CO2 = "SF6") ->
+      EDGAR_SF6
+
+    EDGAR_C2F6 %>%
+      mutate(Non.CO2 = "C2F6") ->
+      EDGAR_C2F6
+
+    EDGAR_CF4 %>%
+      mutate(Non.CO2 = "CF4") %>%
+      bind_rows(EDGAR_SF6, EDGAR_C2F6,.) ->
+      ALL_EDGAR_HFC
+
+    Other_F %>%
+      rename(Non.CO2 = Gas) ->
+      Other_F
+
+    # Then, prepare EDGAR data for use.
+    # Map in region ID and sector name, and remove year 1970 to match "HISTORICAL_YEARS" constant.
+
+    ALL_EDGAR_HFC %>%
+      left_join_error_no_match(EDGAR_sector, by = "IPCC_description") %>%
+      standardize_iso(col = "ISO_A3") %>%
+      change_iso_code('rou','rom') %>%                                        # Switch Romania iso code to its pre-2002 value
+      left_join_error_no_match(iso_GCAM_regID, by = "iso")  %>%
+      rename(EDGAR_agg_sector = agg_sector) %>%
+      select(GCAM_region_ID, EDGAR_agg_sector, Non.CO2, year, value)  %>%
+      filter(year !="1970") ->
```

Comment:
Yes, but I get a stern warning message about object length.  It is because the input data is 1970:2008.  HISTORICAL_YEARS is 1971:2010.  I thought it was cleaner to just take out the "1970".  Red warning messages give me heart palpitations!